pennylane/transforms/optimization/single_qubit_fusion.py

```diff
@@ -14,19 +14,236 @@
+
+            return self._handle_fusible_op(op, cumulative_angles)
+
+        def interpret_all_previous_ops(self) -> None:
+            """Interpret all previous operations stored in the instance."""
+
+            for op in self.previous_ops.values():
+                super().interpret_operation(op)
+
+            self.previous_ops.clear()
+
+        def eval(self, jaxpr: "jax.core.Jaxpr", consts: list, *args) -> list:
+            """Evaluate a jaxpr.
+
+            Args:
+                jaxpr (jax.core.Jaxpr): the jaxpr to evaluate
+                consts (list[TensorLike]): the constant variables for the jaxpr
+                *args (tuple[TensorLike]): The arguments for the jaxpr.
+
+            Returns:
+                list[TensorLike]: the results of the execution.
+            """
+
+            self.setup()
+
+            for arg, invar in zip(args, jaxpr.invars, strict=True):
+                self._env[invar] = arg
+            for const, constvar in zip(consts, jaxpr.constvars, strict=True):
+                self._env[constvar] = const
+
+            for eqn in jaxpr.eqns:
+
+                prim_type = getattr(eqn.primitive, "prim_type", "")
+
+                custom_handler = self._primitive_registrations.get(eqn.primitive, None)
+                if custom_handler:
+                    self.interpret_all_previous_ops()
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    outvals = custom_handler(self, *invals, **eqn.params)
+                elif prim_type == "operator":
+                    outvals = self.interpret_operation_eqn(eqn)
+                elif prim_type == "measurement":
+                    self.interpret_all_previous_ops()
+                    outvals = self.interpret_measurement_eqn(eqn)
+                else:
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    subfuns, params = eqn.primitive.get_bind_params(eqn.params)
+                    outvals = eqn.primitive.bind(*subfuns, *invals, **params)
+
+                if not eqn.primitive.multiple_results:
+                    outvals = [outvals]
+                for outvar, outval in zip(eqn.outvars, outvals, strict=True):
+                    self._env[outvar] = outval
+
+            self.interpret_all_previous_ops()
+
+            outvals = []
+            for var in jaxpr.outvars:
+                outval = self.read(var)
+                if isinstance(outval, Operator):
+                    outvals.append(super().interpret_operation(outval))
+                else:
+                    outvals.append(outval)
+
+            self.cleanup()
+            return outvals
+
+    def single_qubit_fusion_plxpr_to_plxpr(
+        jaxpr, consts, targs, tkwargs, *args
+    ):  # pylint: disable=unused-argument
+        interpreter = SingleQubitFusionInterpreter(
+            atol=tkwargs.pop("atol", 1e-8), exclude_gates=tkwargs.pop("exclude_gates", None)
+        )
```

Comment:
I think we want to use `get` here instead of `pop`.
```suggestion
        interpreter = SingleQubitFusionInterpreter(
            atol=tkwargs.get("atol", 1e-8), exclude_gates=tkwargs.get("exclude_gates", None)
        )
```

Reply:
Yes I'll go over all the transforms to implement this change + `kwargs` validation in a separate PR, but I don't see why we shouldn't use `get` here already at this stage